apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java

```diff
@@ -804,22 +817,38 @@ public ToProtoValueCodec getValueCodec() {
 
     @Override
     public QueryOuterClass.Value protoValueFromStrictlyTyped(Object mapValue) {
-      if (mapValue instanceof Map<?, ?>) {
-        // Maps are actually encoded as Collections where keys and values are interleaved
-        List<QueryOuterClass.Value> elements = new ArrayList<>();
-        for (Map.Entry<?, ?> entry : ((Map<?, ?>) mapValue).entrySet()) {
-          elements.add(keyCodec.protoValueFromStrictlyTyped(entry.getKey()));
-          elements.add(valueCodec.protoValueFromStrictlyTyped(entry.getValue()));
+      if (compactMapData) {
+        if (mapValue instanceof Map<?, ?>) {
+          // Maps are actually encoded as Collections where keys and values are interleaved
+          List<QueryOuterClass.Value> elements = new ArrayList<>();
+          for (Map.Entry<?, ?> entry : ((Map<?, ?>) mapValue).entrySet()) {
+            elements.add(keyCodec.protoValueFromStrictlyTyped(entry.getKey()));
+            elements.add(valueCodec.protoValueFromStrictlyTyped(entry.getValue()));
+          }
+          return Values.of(elements);
+        }
+      } else {
+        if (mapValue instanceof List<?>) {
+          List<QueryOuterClass.Value> elements = new ArrayList<>();
+          for (Object entry : (List<?>) mapValue) {
+            Map<?, ?> entryNode = (Map<?, ?>) entry;
+            elements.add(keyCodec.protoValueFromStrictlyTyped(entryNode.get("key")));
```

Comment:
I think code here should validate, strictly that the entry has exactly two entries; one for `key`, one for `value`, and nothing else; and throw appropriate exceptions for missing or extra entries.
This would make debugging easier, otherwise I think we'd just get NPEs if one or both missing, and extra values silently ignored.

Reply:
Agreed. Handled and added relevant test cases as well in this commit [aebff81](https://github.com/stargate/stargate/pull/2651/commits/aebff812897456ac0c5008df9385a12d81ad0465).